### PR TITLE
Remove view box

### LINKED ISSUE #100: Huge fonts when opening floorplan
I saved a floorplan and then re-opened it. It looks something like this:

image

Floor plan JSON file:

[floorplan.json.txt](https://github.com/NREL/openstudio-geometry-editor/files/1142858/floorplan.json.txt)

### PR DESCRIPTION
**Pull request by bgschiller**
fixes #100, #174, and #132.

This is the longer-term fix whose arrival was foretold in #184 🔮  👻 .

The main point of this PR is removing this line:
```
this.$refs.grid.setAttribute('viewBox', `0 0 ${this.max_x - this.min_x} ${this.max_y - this.min_y}`);
```
Almost everything else was removing the workarounds that were added to accommodate that line, or adding things to correct for the fact that we're now working on the scale of pixels instead of real-world-units.

That line said "scale everything inside the svg so that when I say "1" I mean "1 meter" or "1 foot", instead of "1 pixel" as would be the default. That meant:

- font sizes now needed to be scaled. "12" wasn't "12pt", it was "12m", so they were scaled down to avoid taking up the whole screen. This didn't always work, as in #100.
- When working in a small region, say, `max_x - min_x` less than 20 or so, there was a noticeable offset between where the grid was drawn, and where click events registered. This is because d3 applies a 0.5 offset when drawing axes (https://groups.google.com/forum/#!topic/d3-js/-Sw0Wdnkeko) in order to fix a rendering bug in safari. 0.5 was intended to mean '0.5pixels', but was being interpreted as '0.5 ft'.

Also in this PR, I changed the grid and polygon drawing code to update existing DOM nodes if possible, rather than destroying and creating new ones. Not sure if that actually improves things, since vue was going to do a virtual dom diff anyway. ¯\\\_(ツ)_/¯

### CODE REVIEW

File: src/store/modules/project/index.js [hunk 1]
```diff
@@ -13,14 +13,14 @@ export default {
     },
     grid: {
       visible: true,
-      spacing: 50
+      spacing: 50,
```
Comment (gh-23378) [inline src/store/modules/project/index.js]: let's decrease the default spacing and update our y values for when the grid loads

File: src/components/Grid/Grid.vue [hunk 2]
```diff
@@ -53,6 +52,7 @@ export default {
 
     // add event listeners
     this.$refs.grid.addEventListener('reloadGrid', this.renderGrid);
+    this.$refs.grid.addEventListener('reloadGridScales', this.reloadGridAndScales);
```
Comment (gh-23378) [inline src/components/Grid/Grid.vue]: when you zoom too far out, disable the grid for performance

File: src/components/Grid/Grid.vue [hunk 3]
```diff
@@ -223,17 +225,17 @@ export default {
     },
     transform(newTransform, lastTransform) {
       // hide grid if zoomed out enough
-      this.forceGridHide = (newTransform.kAbs < 0.1);
+      this.forceGridHide = (newTransform.k < 0.1);
```
Comment (gh-23378) [inline src/components/Grid/Grid.vue]: oops, didn't see this. looks like we're already disabling the grid but there's something stopping it from happening

File: src/store/utilities/importFloorplan.js [hunk 4]
```diff
@@ -79,5 +79,5 @@ export default function importFloorplan(context, payload) {
     geometry,
   });
 
-  document.getElementById('svg-grid').dispatchEvent(new Event('reloadGrid'));
+  document.getElementById('svg-grid').dispatchEvent(new Event('reloadGridScales'));
```
Comment (gh-23378) [inline src/store/utilities/importFloorplan.js]: having a strange effect on image imports

**Review by gh-23378 — CHANGES_REQUESTED** [review]

**Comment by bgschiller** [comment]
@devKate Images are working very well now. Lots of changes, so it's probably worth another review.

**Comment by gh-23378** [comment]
Nice! Looks good :)

**Pull request merged**